service/controller/aws/v14/resource/kubeconfig/current_test.go

```diff
@@ -0,0 +1,100 @@
+package kubeconfig
```

Comment:
Let's remove the current state test. This is just checking we can get a secret which is standard `client-go` stuff.

Having a test for `key.KubeConfigSecretName` and the `GetDesiredState` is enough IMO.